### PR TITLE
Feature/red 301 user UUID

### PR DESCRIPTION
**Pull request by red-tt-bear**
Make sure to review my other pr first

### CODE REVIEW

File: src/Account/AccountDetailsPage.tsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+import localPreferences from '../Utilities/LocalPreferences';
+const { session } = localPreferences;
 
 interface AccountDetailsPageProps {
 
 }
 
 export const AccountDetailsPage: React.FC<AccountDetailsPageProps> = () => {
-    const userName = Cookies.get(CookieEnum.USERNAME);
+    // const userName = Cookies.get(CookieEnum.USERNAME);
```
Comment (PanopticaRising) [inline src/Account/AccountDetailsPage.tsx:12]: nit: Useless comment

File: src/Enums/UserRole.tsx [hunk 2]
```diff
@@ -32,9 +37,9 @@ export const getUserRoleFromServer = (roleFromServer: number): UserRole => {
 };
 
 export const getUserRole = (): UserRole => {
-    const roleFromCookie = Cookies.get(CookieEnum.USERTYPE);
-    // eslint-disable-next-line eqeqeq
-    if (roleFromCookie == undefined) {
+    // const roleFromCookie = Cookies.get(CookieEnum.USERTYPE);
+    const roleFromCookie = session.userType;
+    if (roleFromCookie === null) {
```
Comment (PanopticaRising) [inline src/Enums/UserRole.tsx:42]: Should just use `isNil` at this point

File: src/Enums/UserRole.tsx [hunk 3]
```diff
@@ -55,10 +60,11 @@ export const getUserRole = (): UserRole => {
 };
 
 export const getUserId = () => {
-    const userId = Cookies.get(CookieEnum.USERID);
+    // const userId = Cookies.get(CookieEnum.USERID);
+    const userId = session.userId;
     let userIdValue = 0; 
 
-    if (userId === undefined || isNaN(userIdValue = parseInt(userId, 10))) {
+    if (userId === null || isNaN(userIdValue = parseInt(userId, 10))) {
```
Comment (PanopticaRising) [inline src/Enums/UserRole.tsx:67]: Unrelated, but I don't see any reason for the parseInt to be in the conditional here.

File: src/Login/LoginForm.tsx [hunk 4]
```diff
@@ -55,18 +55,23 @@ export const LoginForm: React.FC<LoginFormProps> = () => {
...
                 setLoginAlertMsg({ message: resp.data?.msg || 'Logged in!', variant: 'success' });
-                // TODO: Create a User class to massage and error-handle these fields.
-                Cookie.set(CookieEnum.USERTYPE, getUserRoleFromServer(resp.data.data.roleId));
-                Cookie.set(CookieEnum.USERID, resp.data.data.userId);
-                Cookie.set(CookieEnum.USERNAME, `${resp.data.data.firstName} ${resp.data.data.lastName}`);
+                session.userId = resp.data.data.userId;
+                session.userType = getUserRoleFromServer(resp.data.data.roleId);
+                session.userUUID = resp.data.data.uuid;
+                session.username = `${resp.data.data.firstName} ${resp.data.data.lastName}`;
+                // TODO delete this code, moving to local preferences but want to keep this for debugging
```
Comment (PanopticaRising) [inline src/Login/LoginForm.tsx:62]: You can delete this now.

**Review by PanopticaRising — APPROVED** [review]

**Review by doombot-exe — APPROVED** [review]

File: src/Enums/UserRole.tsx (hunk 3 above)
Comment (red-tt-bear) [reply src/Enums/UserRole.tsx:67]: Yeah, when I clean up I will move this into the local storage fetch like I do elsewhere

File: src/Login/LoginForm.tsx (hunk 4 above)
Comment (red-tt-bear) [reply src/Login/LoginForm.tsx:62]: Saving it since this is a big change, I'll clean it up in dev after it is tested in stage

**Pull request merged**